server/routerlicious/.changeset/polite-brooms-grab.md

```diff
@@ -0,0 +1,11 @@
+---
+"@fluidframework/server-routerlicious-base": major
+"@fluidframework/server-services-shared": major
+---
+---
+"section": fix
```

Comment:
I agree this is technically a fix (and yay for it! It always bugged me that everything was a 500 😆 ) but at this point do we need to consider it a breaking change? I'm a bit concerned about the client behaving in the same way if it starts seeing different error codes for something that has been stable for...ever?

Reply:
Based on https://github.com/microsoft/FluidFramework/pull/23681#discussion_r1934491213 I guess it might not as big a change as I thought...? We still expect all APIs except getDeltas to produce the same error codes they were producing before? Even if some of them (I think, from memory of having seen some of them many months ago) might not technically correct.

Reply:
> I'm a bit concerned about the client behaving in the same way if it starts seeing different error codes for something that has been stable for...ever?

While it has been "stable" forever (i.e. indefinitely retrying regardless of server error), I don't think it has been optimal. Every other API in R11s sends the client accurate error codes (e.g. 400, 404, 401, 403, 500, 503) and the client handles those fine. Personally, I'm not concerned about this change.

Reply:
Exactly. This should not affect any cases other than a small subset of getDeltas even though handleResponse is used widely.